src/App.jsx

```diff
@@ -14,33 +14,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
-
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
-
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
-
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
-
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortBy, setSortBy] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const displayed = useMemo(() => {
+    const result = [...goodsFromServer];
+
+    if (sortBy === 'alpha') {
+      result.sort((a, b) => a.localeCompare(b));
+    } else if (sortBy === 'length') {
+      result.sort((a, b) => a.length - b.length || a.localeCompare(b));
+    }
+
+    if (isReversed) {
+      result.reverse();
+    }
+
+    return result;
+  }, [sortBy, isReversed]);
+
+  const isInitialOrder = useMemo(() => {
+    return (
+      displayed.length === goodsFromServer.length &&
+      displayed.every((g, i) => g === goodsFromServer[i])
+    );
+  }, [displayed]);
```

Comment:
You don't need this variable. You can just check your states, to see if Reset button should be active